### DISCUSSION TITLE
How to Do Translations of UI

### DISCUSSION THREAD

**Discussion by exeqtwasnotfree** [atrocore/atrocore#286] [Q&A] [answered]
Hi, sorry for beginner question.
I´m wondering if there is work being done to translate the interface/locales to more languages?
I looked around in the .json-files on disk and found the 3-4 standard languages, seems to be manageable to make copy's end translate?
Can i somehow help translate to language sv_SE from the en_US files?
How to get that info into Atrocore /AtroPIM?
I´m new to GitHub, so i don´t know how i can upload files, but i want to help if a can.

**Comment by atrolex** [comment]
Hey @exeqtwasnotfree, It's great, if want to help :) All translations are stored in JSONs.
We can proceed as follows, we copy translations in English for Swedish. You can translate than everything and make a pull request (if you can, otherwise you can just upload new files as archive here). Should we start?

**Reply by exeqtwasnotfree** [reply]
Sound very interesting. And i guess when you say copy English to Swedish means that Swedish appear in Atrocore as a selectable language, and new files exist on github ready to be translated, and that English is the fall-back language if not translated in the new files?
Have i got that right?

I´m ready to give it a go.
Company i work for are testing this system, and we like it a lot so far, and translate it to Swedish would make it even better for us. 
So i may be getting help translating and review  if i´m lucky :-)

**Comment by atrolex** [comment, marked as answer]
@exeqtwasnotfree we discussed it internally, the best way to do it via the standard functionality, not by coping files..

So,
1) go to this page and add your language to the List View: https://demo.atropim.com/#Admin/layouts/scope=Translation&type=list

2) than do the same for Detail View: https://demo.atropim.com/#Admin/layouts/scope=Translation&type=detail
Do the same for List View (small) and Detail View (small)

3) Do you translations via Label Manager: https://demo.atropim.com/#Translation
> You can use filters and all other standard functionality here.

4) After you are done, send us your translations for the review:
![image](https://user-images.githubusercontent.com/33658481/153609656-1125e566-a659-4d2f-962f-9845745be2b9.png)

5) Go to Administration > System > Settings and activate your language:
![image](https://user-images.githubusercontent.com/33658481/153609836-109587ff-1cc2-4edd-8aef-09164f752fea.png)
Remove all other languages you don't need.

6) Use the system in your language (you don't need to wait till we accept your translations)

**Exchange demo.atropim.com with your URL!**

Yes, English is the Fallback Language.

**Reply by atrolex** [reply]
Share your feedback about the usability of this feature :)

**Reply by exeqtwasnotfree** [reply]
Very nice. 
I had found the translation-menu before and assumed that it didn´t show more languages do to missing files.
But this insight made me learn more about the system and its configuration. 
And by this way you can set up the system peppered to your language of choice from start, and translate it bit by bit if you can´t manage all in one go. 
For me it means we can continue testing and implementing the system parallel to translating, (no stopping and waiting for translating).
But i figure we only "Send for Review" when we have all or almost all translated?
Or do you want it in batches of 200 ;-)
All and all, this gonna be fun to test this for real first thing Monday morning, looking good so far.

Thank you 4lexo for the support!

**Reply by atrolex** [reply]
It is better to send translations only once, Good Luck with it!

**Comment by exeqtwasnotfree** [comment]
We have now translated little more than 3100 labels to Swedish, and have sent this in for review.
Hope you received it.
And the list with "language (country) › la_CA" there may be some changes that has to be made there,  for example "Serbia and Montenegro" is split up now a days.

**Comment by atrolex** [comment]
Hi @exeqtwasnotfree we will release it in the next version. Thank you very much for the translation :)
